rdt/transformers/categorical.py

```diff
@@ -223,28 +223,54 @@ class OneHotEncodingTransformer(BaseTransformer):
     dummy_na = None
     dummies = None
 
+    @staticmethod
+    def _prepare_data(data):
+        """Transform data to appropriate format.
+
+        If data is a valid list or a list of lists, transforms it into an np.array,
+        otherwise returns it.
+
+        Args:
+            data (pandas.Series, numpy.ndarray, list or list of lists):
+                Data to prepare.
+
+        Returns:
+            pandas.Series or numpy.ndarray
+        """
+        if isinstance(data, list):
+            data = np.array(data)
+            if len(data.shape) == 2:
```

Comment:
Maybe we should put this block one level higher, outside of the `isinstance(data, list)` block?

This way we would also support 2D numpy arrays as input.

Reply:
Sorry, I'm a little confused here, you mean just delete the isinstance?

Reply:
No, I mean removing the indentation of the `if len(data.shape)` lines one level, so they are always run independently on whether the input was a list or a numpy array.

This way, if the input was already a numpy array, its shape is validated too.

Reply:
Oh, mean this?

```python3
if isinstance(data, list):
    data = np.array(data)

if isinstance(data, np.ndarray):
    if len(data.shape) == 2:
        if data.shape[1] != 1:
            raise ValueError("Unexpected format.")
    
        data = data[:, 0]

    elif len(data.shape) > 2:
        raise ValueError("Unexpected format.")
    
return data
```

Maybe at this point we might as well just support anything np.ndarray does? Like this:

```python3
data = np.array(data)
if len(data.shape) == 2:
    if data.shape[1] != 1:
        raise ValueError("Unexpected format.")

    data = data[:, 0]

elif len(data.shape) > 2:
    raise ValueError("Unexpected format.")

return data
```